01_Libraries/ghost_planners/include/ghost_planners/robot_trajectory.hpp

```diff
@@ -11,67 +11,84 @@ namespace ghost_planners {
 class RobotTrajectory {
 public:
 
-	struct MotorTrajectory {
-		MotorTrajectory(){
+	struct Trajectory {
+		Trajectory(){
 			time_vector = std::vector<double>();
 			position_vector = std::vector<double>();
 			velocity_vector = std::vector<double>();
-			voltage_vector = std::vector<double>();
-			torque_vector = std::vector<double>();
-		}
 
-		std::tuple<bool, double> getPosition(double time) const {
+			threshold = 1.0;
+			// voltage_vector = std::vector<double>();
+			// torque_vector = std::vector<double>();
+		}
+		double getPosition(double time) const {
+			return ghost_util::clampedLinearInterpolate(time_vector, position_vector, time);
+		}
+		double getVelocity(double time) const {
+			if (!time_vector.empty() && time > *time_vector.end()){
+				return 0.5 * *velocity_vector.end();
+			}
+			return ghost_util::clampedLinearInterpolate(time_vector, velocity_vector, time);
+		}
+		std::tuple<bool, double> getPositionAndCheck(double time) const {
 			if(position_vector.size() == 0){
 				return {false, 0};
 			}
 			return {true, ghost_util::clampedLinearInterpolate(time_vector, position_vector, time)};
 		}
-		std::tuple<bool, double> getVelocity(double time) const {
+		std::tuple<bool, double> getVelocityAndCheck(double time) const {
```

Comment:
Usually a method like this returns a bool only, and passes whatever value you want by reference

so instead of
`tuple<bool,` double> doThing(double time)`

it would be
`bool doThing(double time, double& velocity)`